### PR TITLE
xdg-desktop-portal-wlr: update expression and unpin

### PR DESCRIPTION
**Pull request by Artturin**
https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/libraries/xdg-desktop-portal-gtk/default.nix

### CODE REVIEW

**Comment by berbiche** [comment]
While at it, can you also add `meta.mainPrograms = "xdg-desktop-portal-wlr"` per https://github.com/colemickens/nixpkgs-wayland/issues/282#issuecomment-789457687?

File: pkgs/xdg-desktop-portal-wlr/default.nix [hunk 1]
```diff
@@ -18,11 +16,16 @@ stdenv.mkDerivation rec {
...
+  buildInputs = [ pipewire wayland systemd libdrm iniparser scdoc inih ];
 
-  enableParallelBuilding = true;
+  mesonFlags = [
+    "-Dsd-bus-provider=libsystemd"
+  ];
+
+  postInstall = ''
+    wrapProgram $out/libexec/xdg-desktop-portal-wlr --prefix PATH ":" ${lib.makeBinPath [ grim slurp ]}
+  '';
```
Comment (berbiche) [inline pkgs/xdg-desktop-portal-wlr/default.nix:28]: And I suppose this will enable the use of the new display "chooser" added upstream?

Comment (Artturin) [reply pkgs/xdg-desktop-portal-wlr/default.nix:28]: yes slurp is the selector

**Comment by Artturin** [comment]
> `meta.mainPrograms = "xdg-desktop-portal-wlr"`

the executable for this is in /libexec/

**Comment by berbiche** [comment]
> > `meta.mainPrograms = "xdg-desktop-portal-wlr"`
> 
> the executable for this is /libexec/xdg-desktop-portal-wlr so does this still need to be added?

Oops, nevermind in that case

**Comment by colemickens** [comment]
Hm, just to check, in the OP you linked to the GTK portal package? But the diff aligns with what I'd expect to see for the wlr portal changes to bring them in alignment with upstream nixpkgs.

Did you just grab the wrong link presumably?

**Comment by Artturin** [comment]
yep i did https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/libraries/xdg-desktop-portal-wlr/default.nix

**Comment by colemickens** [comment]
Okay. Patch builds for me, good enough for me to merge. Thanks!

**Pull request merged**